py-polars/polars/expr/datetime.py

```diff
@@ -2142,35 +2142,221 @@ def month_end(self) -> Expr:
+        │ 2000-03-15 02:00:00 ┆ 2000-03-31 02:00:00 │
+        │ 2000-04-15 02:00:00 ┆ 2000-04-30 02:00:00 │
+        │ 2000-05-15 02:00:00 ┆ 2000-05-31 02:00:00 │
+        │ …                   ┆ …                   │
+        │ 2000-08-15 02:00:00 ┆ 2000-08-31 02:00:00 │
+        │ 2000-09-15 02:00:00 ┆ 2000-09-30 02:00:00 │
+        │ 2000-10-15 02:00:00 ┆ 2000-10-31 02:00:00 │
+        │ 2000-11-15 02:00:00 ┆ 2000-11-30 02:00:00 │
+        │ 2000-12-15 02:00:00 ┆ 2000-12-31 02:00:00 │
+        └─────────────────────┴─────────────────────┘
         """
         return wrap_expr(self._pyexpr.dt_month_end())
 
+    def quarter_start(self) -> Expr:
+        """
+        Roll backward to the first day of the quarter.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`Date` or :class:`Datetime`.
+
+        Notes
+        -----
+        If you're coming from pandas, you can think of this as a vectorised version
+        of `pandas.tseries.offsets.QuarterBegin(startingMonth=1).rollback(datetime)`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "dates": pl.datetime_range(
+        ...             datetime(2000, 1, 15, 2),
+        ...             datetime(2000, 12, 15, 2),
+        ...             "1mo",
+        ...             eager=True,
+        ...         )
+        ...     }
+        ... )
+        >>> df.with_columns(dates_quarter_start=pl.col("dates").dt.quarter_start())
+        shape: (12, 2)
+        ┌─────────────────────┬─────────────────────┐
+        │ dates               ┆ dates_quarter_start │
+        │ ---                 ┆ ---                 │
+        │ datetime[μs]        ┆ datetime[μs]        │
+        ╞═════════════════════╪═════════════════════╡
+        │ 2000-01-15 02:00:00 ┆ 2000-01-01 02:00:00 │
+        │ 2000-02-15 02:00:00 ┆ 2000-01-01 02:00:00 │
+        │ 2000-03-15 02:00:00 ┆ 2000-01-01 02:00:00 │
+        │ 2000-04-15 02:00:00 ┆ 2000-04-01 02:00:00 │
+        │ 2000-05-15 02:00:00 ┆ 2000-04-01 02:00:00 │
+        │ …                   ┆ …                   │
+        │ 2000-08-15 02:00:00 ┆ 2000-07-01 02:00:00 │
+        │ 2000-09-15 02:00:00 ┆ 2000-07-01 02:00:00 │
+        │ 2000-10-15 02:00:00 ┆ 2000-10-01 02:00:00 │
+        │ 2000-11-15 02:00:00 ┆ 2000-10-01 02:00:00 │
+        │ 2000-12-15 02:00:00 ┆ 2000-10-01 02:00:00 │
+        └─────────────────────┴─────────────────────┘
+        """
+        return wrap_expr(self._pyexpr.dt_quarter_start())
+
+    def quarter_end(self) -> Expr:
+        """
+        Roll forward to the last day of the quarter.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`Date` or :class:`Datetime`.
+
+        Notes
+        -----
+        If you're coming from pandas, you can think of this as a vectorised version
+        of `pandas.tseries.offsets.QuarterEnd().rollforward(datetime)`.
```

Comment:
lol wtf pandas QuarterEnd starts in January but QuarterBegin starts in December? 🤯 

(I'm not asking you to do anything here, I'm just...amazed)